src/main/java/ui/GUIController.java

```diff
@@ -102,11 +125,54 @@ public void panelFilterExpressionChanged(FilterPanel changedPanel) {
-                                                             e.remainingRequests,
-                                                             Utility.minutesFromNow(e.nextRefreshInMillisecs)))
-        );
+    /**
+     * Updates UI components using the API rate limits information.
+     * @param e The current API rate limits
+     */
+    private void updateRateLimits(RateLimitsUpdatedEvent e) {
+        updateAPIBox(e.remainingRequests, e.nextRefreshInMillisecs);
+    }
+
+    /**
+     * Updates the period of the refresh timer for synchronization of the data store.
+     * @param e The current API rate limits for calculation of the refresh rate.
+     */
+    private void updateSyncRefreshRate(RefreshTimerTriggeredEvent e) {
+        apiCallsUsedInPreviousRefresh = computePreviousRemainingApiRequests(e.remainingRequests);
+        refreshDurationInMinutes = RefreshTimer.computeRefreshTimerPeriod(e.remainingRequests,
+                                                                    Utility.minutesFromNow(e.nextRefreshInMillisecs),
+                                                                    apiCallsUsedInPreviousRefresh,
+                                                                    RefreshTimer.API_QUOTA_BUFFER,
+                                                                    RefreshTimer.DEFAULT_REFRESH_PERIOD_IN_MINS);
+        ui.refreshTimer.changeRefreshPeriod((int) refreshDurationInMinutes);
+        logger.info("Refresh period updated to " + refreshDurationInMinutes
+                    + "mins with apiCalls used in previous refresh cycle is " + apiCallsUsedInPreviousRefresh
+                    + ", current api quota is " + e.remainingRequests + " and next quota top-up in "
+                    + Utility.minutesFromNow(e.nextRefreshInMillisecs) + "mins.");
+
+    }
+
+    private int computePreviousRemainingApiRequests(int remainingRequests) {
+        int difference = previousRemainingApiRequests - remainingRequests;
+        previousRemainingApiRequests = remainingRequests;
+
+        if (difference >= 0) {
+            return difference;
+        }
+        return apiCallsUsedInPreviousRefresh;
+    }
+
+    /**
+     * Updates the GUI APIBox to indicate the no of remaining API request, time until next API renewal and
+     * the current sync refresh rate.
+     * @param remainingRequests The number of API requests remaining in the current rate limit window.
+     * @param nextRefreshInMillisecs The time at which the current API rate limit window resets
+     *                               in UTC epoch milliseconds.
+     */
+    private void updateAPIBox(int remainingRequests, long nextRefreshInMillisecs) {
+        Platform.runLater(() -> apiBox.setText(String.format("%s/%s[x%d]", remainingRequests,
+                                               Utility.minutesFromNow(nextRefreshInMillisecs),
+                                               (int) Math.ceil(refreshDurationInMinutes))));
```

Comment:
I prefer if these calculations are done outside of this method. This method is just for updating the GUI, not for calculating anything.